### PR TITLE
CI: migrate remaining misc. bash script tests to run as part of Ginkgo framework

### PR DESCRIPTION
**Pull request by ianvernon**
* Move `tests/00-no-newlines-in-log-call.sh` to `contrib/scripts/no-newlines-in-log-call.sh` and have it run during the `precheck` Makefile target (formerly known as `precheck-gofmt`). 
* Migrate `tests/00-check-examples.sh` to run during Ginkgo tests, which validates all policy examples in certain directories under `examples`. 

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #2535

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

File: test/runtime/Policies.go [hunk 1]
```diff
@@ -1302,3 +1303,101 @@ var _ = Describe("RuntimeValidatedPolicyImportTests", func() {
 		Expect(res.Output().String()).Should(ContainSubstring(allowedVerdict), "Policy trace did not contain %s", allowedVerdict)
 	})
 })
+
+var _ = Describe("RuntimeValidatedPolicyValidationTests", func() {
```
Comment (eloycoto) [inline test/runtime/Policies.go]: Could we change this to a new file called `runtime/Examples` I think that ti's a better place

Comment (ianvernon) [reply test/runtime/Policies.go]: Done.

**Comment by ianvernon** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Comment by ianvernon** [comment]
test-me-please

File: contrib/scripts/no-newlines-in-log-call.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (joestringer) [inline contrib/scripts/no-newlines-in-log-call.sh]: Can we name this something more consistent like `check-log-newlines.sh`?

File: test/runtime/examples.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	initialize := func() {
+		logger = log.WithFields(logrus.Fields{"test": "RuntimeValidatedPolicyValidationTests"})
+		logger.Info("Starting")
+		vm = helpers.CreateNewRuntimeHelper(helpers.Runtime, logger)
+	}
+
+	BeforeEach(func() {
+		once.Do(initialize)
+
```
Comment (joestringer) [inline test/runtime/examples.go]: Extra newline

**Review by joestringer — APPROVED** [review]
LGTM, minor aesthetic comments

File: test/runtime/examples.go (hunk 3 above)
Comment (ianvernon) [reply test/runtime/examples.go]: Fixed.

File: contrib/scripts/no-newlines-in-log-call.sh (hunk 2 above)
Comment (ianvernon) [reply contrib/scripts/no-newlines-in-log-call.sh]: Fixed.

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**